lib/cmds/guide.js

```diff
@@ -31,146 +31,181 @@ inquirer.registerPrompt('directory', selectDirectory)
 
 const guideOptimalColumns = 120
 const GUIDE_MAX_WIDTH = process.stdout.columns < guideOptimalColumns ? process.stdout.columns : guideOptimalColumns
-const installationFolderName = 'contentful-custom-app'
+class AbortedError extends Error { }
+
+const guides = {
+  'starter-nuxt-blog': {
+    name: 'Nuxt Starter Blog',
+    directoryName: 'contentful-nuxt-blog',
+    repository: 'contentful/blog-in-5-minutes',
+    seed: 'blog',
+    setupConfig: async ({cmaToken, cdaToken, spaceId, installationDirectory}) => {
+      const config = await bfj.read(join(installationDirectory, '.contentful.sample.json'))
+      config.CTF_CDA_ACCESS_TOKEN = cdaToken
+      config.CTF_CMA_ACCESS_TOKEN = cmaToken
+      config.CTF_SPACE_ID = spaceId
+      return bfj.write(join(installationDirectory, '.contentful.json'), config, {
+        space: 2
+      })
+    },
+    devExecutable: 'npm',
+    devParameters: ['run', 'dev'],
+    devURI: 'http://127.0.0.1:3000',
+    devBrowserOpenRegex: /DONE/
+  },
+  'starter-gatsby-blog': {
+    name: 'Gatsby Starter Blog',
+    directoryName: 'contentful-gatsby-blog',
+    repository: 'contentful/starter-gatsby-blog',
+    seed: 'blog',
+    setupConfig: async ({cdaToken, spaceId, installationDirectory}) => {
+      const config = await bfj.read(join(installationDirectory, '.contentful.json.sample'))
+      config.spaceId = spaceId
+      config.accessToken = cdaToken
+      return bfj.write(join(installationDirectory, '.contentful.json'), config, {
+        space: 2
+      })
+    },
+    devExecutable: 'npm',
+    devParameters: ['run', 'dev'],
+    devURI: 'http://127.0.0.1:8000',
+    devBrowserOpenRegex: /You can now view .* in the browser./
+  }
+}
 
 export const command = 'guide'
 
 export const desc = 'A guide introducing basic concepts of working with Contentful'
 
-export async function guide () {
+async function loginStep (guideContext) {
   const { cmaToken } = await getContext()
-  let installationDirectory = join(homedir(), installationFolderName)
-  let stepCount = 0
-  log()
-  log(asciiText('Contentful Guide'))
 
-  // Step 1 - Login
   if (!cmaToken) {
+    guideContext.stepCount++
+
+    const {stepCount} = guideContext
```

Comment:
just for my own learning: does assigning here make a copy? If it's just a reference, why not just assign as a `var` in the above line and then use?

Reply:
To be honest: I have no clue 😅